src/transformers/generation_utils.py

```diff
@@ -764,9 +867,31 @@ def greedy_search(
             model_inputs = self.prepare_inputs_for_generation(input_ids, **model_kwargs)
 
             # forward pass to get next token
-            outputs = self(**model_inputs, return_dict=True)
+            outputs = self(
+                **model_inputs,
+                return_dict=True,
+                output_attentions=output_attentions,
+                output_hidden_states=output_hidden_states,
+            )
             next_token_logits = outputs.logits[:, -1, :]
 
+            if output_scores:
+                if logits is None:
```

Comment:
I think we can just do:

`logits += (logits,)` here

Reply:
and no need for a reshape -> IMO it's ok that logits is always of shape `(batch_size, vocab_size)` (don't think we need to add an extra `1` dim in the middle